imagebuildah/build.go

```diff
@@ -1017,7 +1017,19 @@ func (b *Executor) copiedFilesMatch(node *parser.Node, historyTime *time.Time) (
 		// For local files, walk the file tree and check the time stamps.
 		timeIsGreater := false
 		err := filepath.Walk(item, func(path string, info os.FileInfo, err error) error {
-			if info.ModTime().After(*historyTime) {
+			modTime := info.ModTime()
+			// resolve any symlinks that occur to get most up to date modified time.
+			if resolvedPath, err := resolveSymLink(b.contextDir, path); err == nil {
```

Comment:
If the path is a file and not a symlink, does the resolvedPat get set to that file?  A comment in resolveSymlink says it returns "" if a symlink isn't found, but that might be a bad comment.

Reply:
I think you should first check to see if path is a symlink before calling resolveSymLink.

Reply:
Yeah I am working on a patch to better approach this issue. Please ignore this PR till then.